### DISCUSSION TITLE
Welcome to National-University-Ranking-App Discussions!

### DISCUSSION THREAD

**Discussion by TelRich** [TelRich/National-University-Ranking-App#12] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by olufemilite** [comment]
@TelRich Here's the code for my data  cleaning task;
#Changing column headers to lower case and changing spaces to underscore
df_nur.rename(columns={"Name":"name","Location":"location","Rank":"rank","Description":"description","Tuition and fees":"tuition_and_fees","In-state":"in_state","Undergrad enrollment":"undergrad_enrollment"}, inplace=True)
df_nur.head()

**Reply by TelRich** [reply]
Seen. Will implement. Thank you

**Comment by Mueeshassan** [comment]
## States with the highest number of schools
states_nur = nur_data['state_full'].value_counts(ascending=True)
print(states_nur)
print(len(states_nur))
states_nur[states_nur > 10].sum() / states_nur.sum()

ax = states_nur.plot(kind='barh', figsize=(20, 15))
ax.set_xlabel('Number of Schools', fontsize= 30)
ax.set_ylabel('States', fontsize= 30)
## Five states (California, Newyork, Massachusetts, Pennsylvania and Illinois) account for 35% of the total schools on the list.

five_or_more = (states_nur > 3).sum()/len(states_nur)
print(five_or_more)
value_freq = states_nur.value_counts()
print(value_freq)

ax = value_freq.plot(kind='bar')
ax.set_xlabel('Number of Schools')
ax.set_ylabel('Frequency')
## 68% of the states have less than 4 schools on the top 230 list

## Schools in the top 50 and where they are located
nur_top50 = nur_data.head(50)
nur_top50 = nur_top50['state_full'].value_counts(ascending=True)

ax = nur_top50.plot(kind='barh')
ax.set_xlabel('Number of Schools')
ax.set_ylabel('Frequency')
## Schools in California, Massachusetts and New York account for close to half of the top 50 schools in the US

## Checking to see if there is any relationship between the rank column and other numerical columns.
nur_data.corr()
## The result shows the following: 
#### 1. A strong negative relationship between the rank and tuition fees columns
#### 2. A moderate negative relationship between in-state and tuition fees
#### 3. A moderate negative relationship between undergraduate enrollment and tuition fees
#### 4. A moderate positive relationship between undergraduate enrollment and in-state

## Visualizing the rank and tuition fees columns correlation
import matplotlib.pyplot as plt
%matplotlib inline
plt.scatter(nur_data['rank'], nur_data['tuition_and_fees'])

## Visualizing the rank and tuition fees columns correlation
plt.scatter(nur_data['in_state'], nur_data['tuition_and_fees'])

## Visualizing the undergraduate enrollment and tuition fees columns correlation
plt.scatter(nur_data['undergrad_enrollment'], nur_data['tuition_and_fees'])

plt.scatter(nur_data['undergrad_enrollment'], nur_data['in_state'])

## States with the most expensive schools
mean_nur = nur_data.pivot_table(values='tuition_and_fees', index='state_full').sort_values(by='tuition_and_fees', ascending=True)

ax = mean_nur.plot(kind='barh', figsize=(20, 15))
ax.set_xlabel('Average Costs', fontsize= 30)
ax.set_ylabel('States', fontsize= 30)
## The top five states with the most expensive schools are District of Columbia, New Hampshire, Connecticut, Massachusetts and Vermont.
## The five least expensive schools are in South Dakota, Wyoming, Utah, North Dakota and Mississippi

**Reply by TelRich** [reply]
Seen. Will implement

**Reply by TelRich** [reply]
This is a nice insight that we may use when writing the article.

But, i don't think you've solved the task that was assigned to you, here. I don't see the code for that.

**Reply by TelRich** [reply]
Your task is as stated below:

Filter the data frame for the following 5 states. CA, NC, PA, MD, and NH. For each of them, there should be a dataframe sorted by tuition fees, in state, and undergrad enrollment.

I expect to see a code that outputs three data frames for each state. Let me know if you're having difficulties.

**Reply by Mueeshassan** [reply]
ca_state = nur_data[nur_data['state'] == 'CA']
ca_state_tuition = ca_state.sort_values(by='tuition_and_fees', ascending=False)
ca_state_instate = ca_state.sort_values(by='in_state', ascending=False)
ca_state_enrollment = ca_state.sort_values(by='undergrad_enrollment', ascending=False)

nc_state = nur_data[nur_data['state'] == 'NC']
nc_state_tuition = nc_state.sort_values(by='tuition_and_fees', ascending=False)
nc_state_instate = nc_state.sort_values(by='in_state', ascending=False)
nc_state_enrollment = nc_state.sort_values(by='undergrad_enrollment', ascending=False)

pa_state = nur_data[nur_data['state'] == 'PA']
pa_state_tuition = pa_state.sort_values(by='tuition_and_fees', ascending=False)
pa_state_instate = pa_state.sort_values(by='in_state', ascending=False)
pa_state_enrollment = pa_state.sort_values(by='undergrad_enrollment', ascending=False)

md_state = nur_data[nur_data['state'] == 'MD']
md_state_tuition = md_state.sort_values(by='tuition_and_fees', ascending=False)
md_state_instate = md_state.sort_values(by='in_state', ascending=False)
md_state_enrollment = md_state.sort_values(by='undergrad_enrollment', ascending=False)

nh_state = nur_data[nur_data['state'] == 'NH']
nh_state_tuition = nh_state.sort_values(by='tuition_and_fees', ascending=False)
nh_state_instate = nh_state.sort_values(by='in_state', ascending=False)
nh_state_enrollment = nh_state.sort_values(by='undergrad_enrollment', ascending=False)

**Reply by Mueeshassan** [reply]
I am sorry i did the wrong thing. I did not know that there was an instruction for the task. Please let me know if the updated code is good to go

**Reply by TelRich** [reply]
It's fine @Mueeshassan . I'll implement the new code
